opta/core/kubernetes.py

```diff
@@ -175,66 +249,81 @@ def _azure_configure_kubectl(layer: "Layer") -> None:
 
 
 def _aws_configure_kubectl(layer: "Layer") -> None:
-    ensure_installed("aws")
-
-    # Get the current account details from the AWS CLI.
-    try:
-        aws_caller_identity = boto3.client("sts").get_caller_identity()
-    except NoCredentialsError:
-        raise UserErrors(
-            "Unable to locate credentials.\n"
-            "Visit `https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html` "
-            "for more information."
-        )
-    except ClientError as e:
-        raise UserErrors(
-            "The AWS Credentials are not configured properly.\n"
-            f" - Code: {e.response['Error']['Code']} Error Message: {e.response['Error']['Message']}"
-            "Visit `https://docs.aws.amazon.com/sdk-for-java/v1/developer-guide/setup-credentials.html` "
-            "for more information."
-        )
+    config_file_name = (
+        f"{GENERATED_KUBE_CONFIG_DIR}/kubeconfig-{layer.root().name}-{layer.cloud}.yaml"
+    )
+    global GENERATED_KUBE_CONFIG
+    if exists(config_file_name):
+        GENERATED_KUBE_CONFIG = config_file_name
+        return
 
-    current_aws_account_id = aws_caller_identity["Account"]
+    region, account_id = _aws_get_cluster_env(layer.root())
 
     # Get the environment's account details from the opta config
     root_layer = layer.root()
-    env_aws_region, env_aws_account_ids = _aws_get_cluster_env(root_layer)
-
-    # Make sure the current account points to the cluster environment
-    if str(current_aws_account_id) not in env_aws_account_ids:
-        raise UserErrors(
-            fmt_msg(
-                f"""
-                The AWS CLI is not configured with the right credentials to
-                access the {root_layer.name or ""} cluster.
-                ~Current AWS Account ID: {current_aws_account_id}
-                ~Valid AWS Account IDs: {env_aws_account_ids}
-                """
-            )
-        )
-
     cluster_name = get_cluster_name(root_layer)
 
     if cluster_name is None:
         raise Exception(
             "The EKS cluster name could not be determined -- please make sure it has been applied in the environment."
         )
 
-    # Update kubeconfig with the cluster details, and also switches context
-    # TODO: Change AWS to BOTO3
-    nice_run(
-        [
-            "aws",
-            "eks",
-            "update-kubeconfig",
-            "--name",
-            cluster_name,
-            "--region",
-            env_aws_region,
```

Comment:
Similar suggestion as above, we can specify the kubeconfig in the aws eks update-kubeconfig command

https://docs.aws.amazon.com/cli/latest/reference/eks/update-kubeconfig.html

Reply:
Ignore this - as agreed above.